src/client/connect.rs

```diff
@@ -70,22 +58,14 @@ pub enum ConnectError<T> {
     SpawnError,
 }
 
-// ==== impl ConnectExecutor ====
-
-impl<E, T, B> ConnectExecutor<T, B> for E
-where
-    T: AsyncRead + AsyncWrite + Send + 'static,
-    B: Payload + 'static,
-    E: TypedExecutor<MapErr<HyperConnection<T, B>, fn(hyper::Error) -> ()>>,
-{
-}
-
 // ===== impl Connect =====
 
 impl<A, B, C> Connect<A, B, C, DefaultExecutor>
 where
     C: HttpMakeConnection<A>,
-    B: HttpBody,
+    B: HttpBody + Send + 'static,
```

Comment:
hmm is there a reason we need this send and static bound now?

Reply:
This admittedly was a follow-up commit to restrict the constructors of `Connect` into only producing usable `MakeService` instances. It's not really needed and it is more intrusive than the `Background` change, so it can be dropped.

Reply:
However, for usability in generic code, it's better to get the compiler diagnose failures to meet these bounds on the inherent methods, rather than produce a puzzling error message telling only that `Connect` does not implement `Service` without further specifics. I think this should be addressed during the API polishing phase.

Reply:
I have removed this commit.